src/connection-manager/index.js

```diff
@@ -206,6 +210,7 @@ class ConnectionManager extends EventEmitter {
       this.connections.delete(peerId)
       this._peerValues.delete(connection.remotePeer.toB58String())
       this.emit('peer:disconnect', connection)
+      this._maybeConnectN()
```

Comment:
If a disconnect occurs and the previous call to `_maybeConnectN` happens we're going to run two calls in parallel, we should avoid this.

Reply:
you mean the `_maybeConnectN` in start?

Reply:
We can potentially have it running from the start call while trying to dial a bunch of peers and one of them meanwhile disconnected. I can add a verification for wether we are currently running it